aldryn_events/cms_plugins.py

```diff
@@ -92,15 +92,17 @@ def render(self, context, instance, placeholder):
         context['instance'] = instance
         language = self.get_language(context['request'])
         namespace = self.get_namespace(instance)
-        events = (Event.objects.namespace(namespace)
-                               .active_translations(language)
-                               .language(language))
-        events = events.translated(*self.valid_languages)
-        if instance.past_events:
-            events = events.past(count=instance.latest_entries)
+        if instance.language not in self.valid_languages:
+            events = Event.objects.none()
         else:
-            events = events.upcoming(count=instance.latest_entries)
-
+            events = (Event.objects.namespace(namespace)
+                                   .active_translations(language)
```

Comment:
calling ``active_translations`` and then ``translated`` is redundant.
``active_translations()`` calls ``translated()``. I would remove the active translations one.